css/css-transitions/before-load-001.html

```diff
@@ -1,150 +1,46 @@
-                                var current = data.transition.computedStyle(data.property);
-                                assert_not_equals(current, data.transition.to, "must not be target value after start");
-                            },
-                            done: function(test, data, options) {
-                                // make sure the property's value were neither initial nor target while transitioning
-                                test.step(generalParallelTest.assertIntermediateValuesFunc(data, 'transition'));
-                            }
-                        },
-                        // test TransitionEnd events
-                        'events': {
-                            done: function(test, data, options) {
-                                // make sure there were no events on parent
-                                test.step(generalParallelTest.assertExpectedEventsFunc(data, 'container', ""));
-                                // make sure we got the event for the tested property only
-                                test.step(generalParallelTest.assertExpectedEventsFunc(data, 'transition', addVendorPrefix(data.property) + ":" + duration));
-
-                                test.step(function() {
-                                    assert_false(isLoad, "load may not have happened yet")
-                                });
-                            }
-                        }
-                    },
-                    // called once all tests are done
-                    done: generalParallelTest.done
-                });
-            }, false);
-        </script>
-        <img src="/delay/?type=gif&amp;delay=3000" alt="dummy image">
-    </body>
+<head>
+<meta charset="utf-8">
+<title>CSS Transitions Test: Transitioning before load event</title>
+<meta name="assert" content="Test checks that transitions are run even before the load event fires">
+<!-- NOTE: This test covers unspecified behavior and should probably be
+     removed. It this behavior *were* specified, it would probably be
+     specified here: -->
+<link rel="help" title="5. Transition Events" href="https://drafts.csswg.org/css-transitions/#starting">
+<link rel="author" title="Rodney Rehm" href="http://rodneyrehm.de/en/">
+
+<script src="/resources/testharness.js" type="text/javascript"></script>
+<script src="/resources/testharnessreport.js" type="text/javascript"></script>
+<script src="./support/helper.js" type="text/javascript"></script>
+
+</head>
+<body>
+
+<script>
+// Make sure a transition can run before the load event fires.
+
+let loadEventFired = false;
+window.addEventListener('load', () => {
+  loadEventFired = true;
+}, false);
+
+async_test(t => {
+  const div = addDiv(t, { style: 'transition: height .01s linear; ' +
+                                 'height: 1px' });
+  getComputedStyle(div).height;
+  div.style.height = '100px';
+
+  div.addEventListener('transitionend', t.step_func_done(() => {
```

Comment:
Would it not be sufficient to listen to `transitionrun` here? This should be indicative of a transition being able to run.

Reply:
Yeah, I think initially I tried to avoid using transitionrun since it wasn't so widely supported. Later I ended up using it in other tests though so I should probably update this like you suggest.